### PR TITLE
Mark In-Memory/OTLP/Stdout exporters as Stable

### PR DESCRIPTION
**Pull request by reyang**
This has a dependency on the MetricExporter spec, so it should wait for #2150 to be merged.

### CODE REVIEW

**Review by jsuereth — APPROVED** [review]

**Review by cijothomas — APPROVED** [review]

**Review by jmacd — APPROVED** [review]

**Review by mtwo — APPROVED** [review]

File: specification/metrics/sdk_exporters/in-memory.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # OpenTelemetry Metrics Exporter - In-memory
 
-**Status**: [Feature-freeze](../../document-status.md)
+**Status**: [Stable](../../document-status.md)
```
Comment (jack-berg) [inline specification/metrics/sdk_exporters/in-memory.md:3]: What should the behavior of flush be for in-memory metric exporter? In java, [as noted in this ticket](https://github.com/open-telemetry/opentelemetry-java/issues/3943#issue-1067754880), it triggers a collection, which seems incorrect according to the [force flush description](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#forceflush-1).

Comment (reyang) [reply specification/metrics/sdk_exporters/in-memory.md:3]: I can share what .NET did:
1. InMemoryExporter.ForceFlush simply returns true.
2. If the user is reporting measurements and want to make sure the resulting metrics are collected, they should use either of the following way:

    2.1 Call `MeterProvider.ForceFlush`, which in turn will call `MetricReader.Collect` on all MetricReader instances, which will call `InMemoryExporter.ForceFlush`, the callstack looks like:
    ```text
    InMemoryExporter.ForceFlush
    BaseExportingMetricReader.Collect (or PeriodicExportingMetricReader, depending on the configuration)
    MeterProvider.ForceFlush
    ````
    2.2. Call `MetricReader.Collect` on the MetricReader instance that is associated with the InMemoryExporter.

The difference between 2.1 and 2.2: if there are multiple readers/exporters, 2.1 will trigger all of them (except for exporters that only support pull - e.g. Prometheus), 2.2 will only trigger the collect/flush for the specific reader/exporter.

Comment (jack-berg) [reply specification/metrics/sdk_exporters/in-memory.md:3]: Right, that's the behavior I think is correct as well. I think its surprising for the java `InMemoryMetricExporter#flush()` to call collect, and will work to get that sorted out.

**Review by jack-berg — APPROVED** [review]

**Comment by cijothomas** [comment]
> This has a dependency on the MetricExporter spec, so it should wait for #2150 to be merged.

^^ #2150 is not merged.

**Pull request merged**